Client/README.md

```diff
@@ -694,42 +694,184 @@ mine-sensor,id=132-987-655,customer="California Miner",hostname=example.com,sens
 
 ### Handle the Events
 
-#### Handle the Success write
+Events that can be handle by WriteAPI EventHandler are:
+- `WriteSuccessEvent` - for success response from server
+- `WriteErrorEvent` - for unhandled exception from server
+- `WriteRetriableErrorEvent` - for retriable error from server
+- `WriteRuntimeExceptionEvent` - for runtime exception in background batch processing
+
+Number of events depends on number of data point to collect in batch Batch option in WriteAPI (default setting is 1000) - in case 
+of one data point in batch option, event is handled for each point, independently on used writing method (even for mass writing of data like 
+`WriteMeasurements`, `WritePoints` and `WriteRecords`). 
+
+Events can be handled by register `writeApi.EventHandler` or by creating custom `EventListener`:
+
+#### Register EventHandler
 
 ```c#
-//
-// Register event handler
-//
 writeApi.EventHandler += (sender, eventArgs) =>
 {
-    if (eventArgs is WriteSuccessEvent @event)
+    switch (eventArgs)
     {
-        string data = @event.LineProtocol;
-        
-        //
-        // handle success
-        //
+        case WriteSuccessEvent successEvent:
+            string data = @event.LineProtocol;
+            //
+            // handle success response from server
+            // Console.WriteLine($"{data}");
+            //
+            break;
+        case WriteErrorEvent error:
+            string data = @error.LineProtocol;
+            string errorMessage = @error.Exception.Message;
+            //
+            // handle unhandled exception from server
+            //
+            // Console.WriteLine($"{data}");
+            // throw new Exception(errorMessage);
+            //
+            break;
+        case WriteRetriableErrorEvent error:
+            string data = @error.LineProtocol;
+            string errorMessage = @error.Exception.Message;
+            //
+            // handle retrievable error from server
+            //
+            // Console.WriteLine($"{data}");
+            // throw new Exception(errorMessage);
+            //
+            break;
+        case WriteRuntimeExceptionEvent error:
+            string errorMessage = @error.Exception.Message;
+            //
+            // handle runtime exception in background batch processing
+            // throw new Exception(errorMessage);
+            //
+            break;
     }
 };
-```
-
-#### Handle the Error Write
 
-```c#
 //
-// Register event handler
+// Write by LineProtocol
 //
-writeApi.EventHandler += (sender, eventArgs) =>
+writeApi.WriteRecord("influxPoint,writeType=lineProtocol value=11.11" +
+    $" {DateTime.UtcNow.Subtract(EpochStart).Ticks * 100}", WritePrecision.Ns, "my-bucket", "my-org");
+```
+
+#### Custom EventListener
+
+Advantage of using custom Event Listener is possibility of waiting on handled event between different writings.
+
+```csharp
+internal class EventListener
```

Comment:
Please remove the **Custom EventListener** from`README.md`... it is too much complicated. The link to the example will be fine.

```suggestion
```